x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/repository_list/repository_details/repository_details.tsx

```diff
@@ -318,6 +314,72 @@ export const RepositoryDetails: React.FunctionComponent<Props> = ({
     </Fragment>
   );
 
+  const renderCleanup = () => (
+    <>
+      <EuiTitle size="s">
+        <h3>
+          <FormattedMessage
+            id="xpack.snapshotRestore.repositoryDetails.cleanupTitle"
+            defaultMessage="Repository cleanup"
+          />
+        </h3>
+      </EuiTitle>
+      <EuiSpacer size="s" />
+      <EuiText size="s">
+        <p>
+          <FormattedMessage
+            id="xpack.snapshotRestore.repositoryDetails.cleanupRepositoryMessage"
+            defaultMessage="You can clean up a repository to delete any unreferenced data from a snapshot. This
+              may provide storage space savings. Note: If you regularly delete snapshots, this
+              functionality will likely not be as beneficial and should be used less frequently."
+          />
+        </p>
+      </EuiText>
+      {cleanup ? (
+        <>
+          <EuiSpacer size="s" />
+          {cleanup.cleaned ? (
+            <div>
+              <EuiTitle size="xs">
+                <h4>
+                  <FormattedMessage
+                    id="xpack.snapshotRestore.repositoryDetails.cleanupDetailsTitle"
+                    defaultMessage="Details"
+                  />
+                </h4>
+              </EuiTitle>
+              <EuiCodeBlock language="json" inline={false} data-test-subj="cleanupCodeBlock">
+                {JSON.stringify(cleanup.response, null, 2)}
+              </EuiCodeBlock>
+            </div>
+          ) : (
+            <EuiCallOut color="danger" iconType="alert">
```

Comment:
@cuff-links thanks for adding this!

I do see a few issues that I think need to be addressed.

I'm currently seeing this when I test locally:
<img width="528" alt="Screen Shot 2020-01-09 at 9 47 05 AM" src="https://user-images.githubusercontent.com/5226211/72077405-378aa280-32c5-11ea-9245-6bb7286ab849.png">

1. I think the first string should be passed in as the `title` prop, so it looks more like this:

<img width="1003" alt="Screen Shot 2020-01-09 at 9 48 29 AM" src="https://user-images.githubusercontent.com/5226211/72077407-378aa280-32c5-11ea-92a6-276ca2f15ab7.png">

2. It looks like `error` is an object, so I don't think you can call `toString()`.

3. `'503: Unknown error'` needs i18n

Reply:
Done.